flink-table-store-core/src/main/java/org/apache/flink/table/store/file/mergetree/SortBufferMemTable.java

```diff
@@ -166,19 +169,31 @@ private void advanceIfNeeded() {
                 if (previousRow == null) {
                     return;
                 }
-                mergeFunction.reset();
-                mergeFunction.add(previous.getReusedKv().value());
 
+                boolean hasOne = true;
                 while (readOnce()) {
                     if (keyComparator.compare(
                                     previous.getReusedKv().key(), current.getReusedKv().key())
                             != 0) {
                         break;
                     }
+                    // avoid merging when there is only one record
+                    if (hasOne) {
+                        mergeFunction.reset();
+                        mergeFunction.add(previous.getReusedKv().value());
+                        hasOne = false;
+                    }
                     mergeFunction.add(current.getReusedKv().value());
                     swapSerializers();
                 }
-                result = mergeFunction.getValue();
+                RowData previousValue = previous.getReusedKv().value();
+                result =
+                        hasOne
+                                ? mergeFunction instanceof ValueCountMergeFunction
```

Comment:
I think we can ignore that count is zero for `ValueCountMergeFunction`.
Because if there is no merge, the count should never be zero.

Reply:
@JingsongLi, in theory the count should never be zero, but in current tests there is 0 test case. Hence I add this logic to avoid the zero situation.